crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +339,712 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
+            F,
+            NovalnetCaptureResponse,
+            PaymentsCaptureData,
+            PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        match item.response.result.status {
+            NovalnetAPIStatus::SUCCESS => {
+                let transaction_id = item
+                    .response
+                    .transaction
+                    .clone()
+                    .and_then(|data| data.tid.map(|tid| tid.to_string()));
+                let transaction_status = item
+                    .response
+                    .transaction
+                    .as_ref()
+                    .and_then(|data| {
+                        if data.status_code == Some(100) {
+                            // TODO: verify status_code
+                            data.status.clone()
+                        } else {
+                            None
+                        }
+                    })
+                    .unwrap_or(NovalnetTransactionStatus::PENDING);
+
+                Ok(Self {
+                    status: common_enums::AttemptStatus::from(transaction_status),
+                    response: Ok(PaymentsResponseData::TransactionResponse {
+                        resource_id: transaction_id
+                            .clone()
+                            .map(ResponseId::ConnectorTransactionId)
+                            .unwrap_or(ResponseId::NoResponseId),
+                        redirection_data: None,
+                        mandate_reference: None,
+                        connector_metadata: None,
+                        network_txn_id: None,
+                        connector_response_reference_id: transaction_id.clone(),
+                        incremental_authorization_allowed: None,
+                        charge_id: None,
+                    }),
+                    ..item.data
+                })
+            }
+            NovalnetAPIStatus::FAILURE => {
+                let response = Err(get_error_response(item.response.result, item.http_code));
+                Ok(Self {
+                    response,
+                    status: enums::AttemptStatus::Failure,
+                    ..item.data
+                })
+            }
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetSyncTransaction {
+    tid: String,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub struct NovalnetSyncRequest {
+    pub transaction: NovalnetSyncTransaction,
+    pub custom: NovalnetCustom,
+}
+
+impl TryFrom<&RefundSyncRouterData> for NovalnetSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let transaction = NovalnetSyncTransaction {
+            tid: item.request.connector_transaction_id.clone(),
+        };
+
+        let custom = NovalnetCustom {
+            lang: item.request.get_browser_info()?.get_language()?,
+        };
```

Comment:
Can we pass default language as English if `browser_info` or `language` is not present?